atomic_reactor/plugins/exit_delete_from_registry.py

```diff
@@ -42,9 +42,38 @@ def __init__(self, tasker, workflow, registries):
 
         self.registries = deepcopy(registries)
 
+    def check_for_orchestrator_build(self):
+        """
+         If we are being called from an orchestrator build, collect the worker
+         node data and recreate the data locally.
+        """
+        if are_plugins_in_order(self.workflow.buildstep_plugins_conf, 'orchestrate_build'):
+            # orchestrator node
+            if not self.workflow.build_result.annotations:
+                return
+
+            builds = self.workflow.build_result.annotations['worker-builds']
+
+            for plat, annotation in builds.items():
+                self.log.debug("build %s has annotation: %s" % (plat, annotation))
+                digests = annotation['all-digests']
+
+                for digest in digests:
+                    registry = None
+                    for reg in self.workflow.push_conf.docker_registries:
+                        if reg.uri == digest['registry']:
+                            registry = reg
+
+                    if not registry:
+                        registry = self.workflow.push_conf.add_docker_registry(digest['registry'])
+
+                    registry.digests[digest['tag']] = ManifestDigest(v2=digest['digest'])
+
     def run(self):
         deleted_digests = set()
 
+        self.check_for_orchestrator_build()
```

Comment:
How about having an attribute called: `self.docker_registries`.
If it's in orchestrator mode, it must be populated from workers metadata, otherwise it's just a reference to `self.workflow.push_conf.docker_registries` ?

The the main code in this plugin can iterate over `self.docker_registries`.